google_guest_agent_packaging_diff_tool.py

```diff
@@ -1,46 +1,111 @@
-    # compare the above control fields
+@click.option('--agent-package', required=True,
+              type=click.Choice(["google-guest-agent",
+                                        "gce-compute-image-packages",
+                                        "google-compute-engine-oslogin",
+                                        "google-osconfig-agent"], case_sensitive=False),
+              help='The package to inspect')
+
+def main(agent_package: str) -> None:
+
+    url_list = archive_dict[agent_package]
+
+    ubuntu_req = requests.get(url_list[0])
+    with open(UBUNTU_CONTROL_FILENAME, "w+") as fp:
+        fp.write(ubuntu_req.text)
+
+    google_req = requests.get(url_list[1])
+    with open(GOOGLE_CONTROL_FILENAME, "w+") as fp:
+        fp.write(google_req.text)
+
+    ubuntu_control_fields = parse_control_file(UBUNTU_CONTROL_FILENAME)
+    upstream_control_fields = parse_control_file(GOOGLE_CONTROL_FILENAME)
+
+    fields_to_compare = ['Build-Depends', 'Depends', 'Recommends',
+                         'Suggests', 'Breaks', 'Conflicts',
+                         'Replaces', 'Provides']
+
+    # Compare the above control fields (`fields_to_compare`)
     for field_name in [field_to_compare for field_to_compare in ubuntu_control_fields.keys() if field_to_compare in fields_to_compare]:
-        print(f'{field_name}')
-        if field_name in upstream_control_fields.keys():
-            # print(f'\tupstream: {upstream_control_fields[field_name]}')
-            relations = deb822.PkgRelation.parse_relations(upstream_control_fields[field_name])
-            print(relations)
-            for relation in relations:
-                for relation_item in relation:
-                    print(f'\tupstream: {relation_item["name"]} {relation_item["version"]}')
-        else:
-            print(f'\tupstream: None')
-        if field_name in upstream_control_fields.keys():
-            # print(f'\tubuntu: {ubuntu_control_fields[field_name]}')
-            relations = deb822.PkgRelation.parse_relations(ubuntu_control_fields[field_name])
-            print(relations)
-            for relation in relations:
-                for relation_item in relation:
-                    print(f'\tubuntu: {relation_item["name"]} {relation_item["version"]}')
-        else:
-            print(f'\tubuntu: None')
+        with warnings.catch_warnings():
+            warnings.filterwarnings("error")
+            print(f'###### Comparing {field_name} ######')
+            if field_name in upstream_control_fields.keys():
+                print(f'Google upstream: {upstream_control_fields[field_name]}')
+                try:
+                    upstream_relations = deb822.PkgRelation.parse_relations(upstream_control_fields[field_name])
+                    pprint.pprint(upstream_relations, indent=4, sort_dicts=False)
+                    print("\n")
+                except Warning:
+                    pass
+            else:
+                print(f'Google Upstream: None\n')
```

Comment:
These print statements could be consolidated within each `if` block.

Reply:
Done, it's in the next upload :smile: Along with a few more formatting changes :eyes: